src/Development/Shake/Internal/Rules/File.hs

```diff
@@ -87,16 +93,32 @@ instance BinaryEx [FileQ] where
 instance Hashable FileA where
     hashWithSalt salt (FileA a b c) = hashWithSalt salt a `xor` hashWithSalt salt b `xor` hashWithSalt salt c
 
+instance Eq FileR where
+   fr1 == fr2 = result fr1 == result fr2
+
+instance Hashable FileR where
+    hashWithSalt salt (FileR f _) = hashWithSalt salt f
```

Comment:
The `Hashable` and `Eq` should be consistent - which they are. But one uses pattern matching and one uses `result` selector - they should probably both be the same style.